File: two sum/two_sum.java
package test;

public class two_sum {
	
	 public static int[] twoSum(int[] nums, int target) {
	     for(int i=0;i<nums.length-1;i++) {
	    	 for(int j=i+1;j<nums.length;j++) {
	        		if(nums[i] + nums[j] == target) {
	        			return new int[] { i, j };
	        		}
	        	}
	     }
	     throw new IllegalArgumentException("No two sum solution");
	 }
	 public static void main(String [] args) {
		 int [] nums = {2, 7, 11, 15};
		 int target = 9;
		 int [] index = twoSum(nums,target);
		 System.out.print(index[0]+"  "+index[1]);
	 }
	

}

File: two sum/two_sum.py
def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    a = {}
    for i, num in enumerate(nums):
        if target - num in a:
            return [a[target - num], i]
        else:
            a[num] = i

nums = [2, 7, 11, 15]
target = 9
print(nums)
print(list(enumerate(nums)))
print(twoSum(nums,target))